addons/l10n_latam_check/models/account_payment.py

```diff
@@ -1,328 +1,252 @@
-                error_message = self.env['res.partner']._build_vat_error_message(
-                    rec.company_id.country_id.code.lower(), rec.l10n_latam_check_issuer_vat, 'Check Issuer VAT')
-                raise ValidationError(error_message)
+    def _get_reconciled_checks_error(self):
+        checks_reconciled = self.l10n_latam_new_check_ids.filtered(lambda x: x.issue_state in ['debited', 'voided'])
+        if checks_reconciled:
+            raise UserError(
+                _("You can't cancel or re-open a payment with checks if some check has been debited or been voided. "
+                  "Checks:\n%s", ('\n'.join(['* %s (%s)' % (x.name, x.issue_state) for x in checks_reconciled])))
+            )
+
+    def action_cancel(self):
+        self._get_reconciled_checks_error()
+        super().action_cancel()
+
+    def action_draft(self):
+        self._get_reconciled_checks_error()
+        super().action_draft()
+
+    def _l10n_latam_check_split_move(self):
+        for payment in self.filtered(lambda x: x.payment_method_code == 'own_checks' and x.payment_type == 'outbound'):
+            if len(payment.l10n_latam_new_check_ids) == 1:
+                liquidity_line = payment._seek_for_lines()[0]
+                payment.l10n_latam_new_check_ids.liquidity_line_id = liquidity_line.id
+                continue
+            move_id = self.env['account.move'].create({
+                'journal_id': payment.journal_id.id,
+                'move_type': 'entry',
+            })
+            liquidity_line = payment._seek_for_lines()[0]
+
+            for check in payment.l10n_latam_new_check_ids:
+                liquidity_amount_currency = -check.amount
+                liquidity_balance = payment.currency_id._convert(
+                    liquidity_amount_currency,
+                    payment.company_id.currency_id,
+                    payment.company_id,
+                    payment.date,
+                )
+                # Checks liquidity line
+                move_line = self.env['account.move.line'].with_context(check_move_validity=False).create({
+                    'name': _('Check %s - ', check.name + liquidity_line.name),
+                    'date_maturity': check.payment_date,
+                    'amount_currency': liquidity_amount_currency,
+                    'currency_id': check.currency_id.id,
+                    'debit': max(0.0, liquidity_balance),
+                    'credit': -min(liquidity_balance, 0.0),
+                    'partner_id': payment.partner_id.id,
+                    'account_id': payment.outstanding_account_id.id,
+                    'move_id': move_id.id,
+                })
+                check.liquidity_line_id = move_line.id
+
+            # inverse liquidity line
+            inverse_liquidity_line = self.env['account.move.line'].with_context(check_move_validity=False).create({
```

Comment:
Why do we have an inverse line for every check?  We could just have one big amount instead.

Reply:
Done